libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -353,11 +353,16 @@ void AP_Mount_Siyi::process_packet()
         _fw_version.camera.major = _msg_buff[_msg_buff_data_start+2];  // firmware major version
         _fw_version.camera.minor = _msg_buff[_msg_buff_data_start+1];  // firmware minor version
         _fw_version.camera.patch = _msg_buff[_msg_buff_data_start+0];  // firmware revision (aka patch)
-        
+
         _fw_version.gimbal.major = _msg_buff[_msg_buff_data_start+6];  // firmware major version
         _fw_version.gimbal.minor = _msg_buff[_msg_buff_data_start+5];  // firmware minor version
         _fw_version.gimbal.patch = _msg_buff[_msg_buff_data_start+4];  // firmware revision (aka patch)
-        
+
```

Comment:
Should there be a timeout on this? What if there is a problem and the camera keeps reporting 0.0.0, perhaps the user needs to know that? This will just wait and wait right?

Reply:
@timtuxworth,

Yes, that's the downside I mention in the PR description above.  I'm not aware of any Siyi cameras that ever do this though so I'm not sure if it is worth the extra code to protect against this.